### ISSUE TITLE
Apple App Store issue: ITMS-90338: Non-public API usage 

### ISSUE THREAD

**Issue by xoox** [juicycleff/flutter-unity-view-widget#407]
**Describe the bug**

Hi, thanks for this remarkable library. Have anyone ever tried to upload app to Apple App Store Connect to test/distribute with TestFlight? The app scanning process failed with `ITMS-90338: Non-public API usage` issue.

**To Reproduce**
1. Open the example project in XCode
2. Change the bundle id to a valid one belonging to your Apple developer account.
3. Choose a valid signing profile of app-store distribution type.
4. Set the code signing identity to `iOS Distribution` if necessary.
5. Build a distribution signed ipa and try to upload it to TestFlight
6. You'll get an email from Apple including the following message

> ITMS-90338: Non-public API usage - The app references non-public selectors in Frameworks/flutter_unity_widget.framework/flutter_unity_widget: appController, rootView. If method names in your source code match the private Apple APIs listed above, altering your method names will help prevent this app from being flagged in future submissions. In addition, note that one or more of the above APIs may be located in a static library that was included with your app. If so, they must be removed. For further information, visit the Technical Support Information at http://developer.apple.com/support/technical/ 

**Expected behavior**

No issue reminder from Apple store connect. The build appears in TestFlight.

**Unity (please complete the following information):**
 - OS: iOS

**Comment by almeynman** [comment]
Having the same problem

**Comment by xoox** [comment]
While waiting for fixes from upstream, we just renamed `appController` and `rootView` to something else both in flutter_unity_widget package and the exported `ios/UnityLibrary` project. Specifically, we've done the following steps.

1. Fork flutter_unity_widget and rename `appController` and `rootView` to something else.
2. In `pubspec.yaml`, change dependency to our local fork of flutter_unity_widget.
3. After Unity build, rename `appController` and `rootView` to the same one of step 1, by apply a patch to the exported projected `ios/UnityLibrary`. This step can be performed in Unity build post-process or `ios/Podfile` phase.

Our issue was temporally fixed by above steps. Good luck to you. @almeynman

**Comment by almeynman** [comment]
@xoox Thanks a lot for the steps, we will try them out

**Comment by andywindowsmac** [comment]
Hi @xoox
I also had that issue, but my problem resolved after I changed there:
<img width="692" alt="image" src="https://user-images.githubusercontent.com/17358112/127586439-828f87cb-318b-43c7-8bd5-8e6375e80eaa.png">

it changed from "Do Not Embed" to "Embed & Sign"
but I am not sure why

**Comment by xoox** [comment]
Hi @andywindowsmac , Thanks for your tips.

I remember that the default setting for UnityFramework.framework was `Embed & Sign`. I changed it to `Do Not Embed` some day for no reason. It was built successfully but would run into failure due to missing Unity related data.

This issue might be reported during this testing period. And we proposed our solution for it. But after some while, `Embed & Sign` was selected correctly without knowing the ITMS-90338 will not show again. So the solution was still there.

@andywindowsmac 's answer reminds me to review the above setting and solution. It's true that ITMS-90338 issue would not exist if UnityFramework.framework is set to `Embed & Sign` properly.

In conclusion, 
* `Embed & Sign` SHOULD be selected for UnityFramework.framework and was already selected as far as I know. This setting SHOULD NOT be changed.
* The solution we proposed is unnecessary.

Thanks to @andywindowsmac 's answer, this issue is clarified and actually a false positive signal due to misconfiguration. I'll close it. Thanks all of you for making it solved.
